x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -88,6 +90,9 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
                         e -> handleException("Secondary authentication", request, channel, e)));
                 }, e -> handleException("Authentication", request, channel, e)));
         } else {
+            HeaderWarning.addWarning("Elasticsearch built-in security features are not enabled, your cluster may be accessible without " +
+                "authentication. Read https://www.elastic.co/guide/en/elasticsearch/reference/" + Version.CURRENT.major + "." +
```

Comment:
Ryan [commented](https://github.com/elastic/elasticsearch/pull/66827#discussion_r594779071) that we avoid putting URLs in warning messages. Should we drop it here as well?

Reply:
I think we need to find a way to be comfortable pointing to docs. It's important for the user experience.
We do it for deprecation warnings, and we should do it here, we just need to make sure that we point to a URL that we're happy to maintain forever.

Reply:
Hopefully, with Security On by Default we are able to get rid of these warning headers, so for this specific URL we may not need to support it forever